routing/fake_ending.cpp

```diff
@@ -0,0 +1,79 @@
+#include "routing/fake_ending.hpp"
+
+#include "routing/edge_estimator.hpp"
+#include "routing/index_graph.hpp"
+#include "routing/world_graph.hpp"
+
+#include "geometry/distance.hpp"
+
+#include "base/math.hpp"
+
+#include <functional>
+
+using namespace routing;
+using namespace std;
+
+namespace
+{
+using EstimatorFn = function<double(m2::PointD const &, m2::PointD const &)>;
```

Comment:
Do you really need function? Why not to:
```c++
template <typename Estimator>
Junction CalcProjectionToSegment(..., Estimator && estimator)
{
  ...
}
```
?
In this case you'll avoid wrapping lambda to function, moreover, the code of estimator will be inlined here.

Reply:
See code below: I need to be able to call CalcProjectionToSegment both when I have IndexGraph + Estimator and when I have WorldGraph (which does not have GetEstimator() or other such method)

Reply:
ok

Reply:
done